### DISCUSSION TITLE
doing it the remeda way

### DISCUSSION THREAD

**Discussion by boneskull** [remeda/remeda#919] [Q&A] [answered]
Hi,

I'm coming from lodash & lodash/fp.  I'm struggling to understand how to write something like this:

```ts
export function toDualCasedObject<const T extends Record<string, unknown>>(
  obj: T,
): DualCasedObject<T> {
  return {
    ...R.mapKeys(obj, (key) => R.toCamelCase(key)),
    ...R.mapKeys(obj, (key) => R.toKebabCase(key)),
  } as DualCasedObject<T>;
}
```

This works, but:

1. I cannot replace the second parameter with just `toCamelCase` because `mapKeys` calls it with two parameters and `toCamelCase` expects one.  I thought `identity` might help me get the first parameter, but I'm not sure how to use it in this context
2. I could use `pipe` here:
    ```ts
    return {
      ...pipe(
        obj,
        mapKeys((key) => toCamelCase(key)),
      ),
      ...pipe(
        obj,
        mapKeys((key) => toKebabCase(key)),
      ),
    } as DualCasedObject<T>;
    ```
    
    Is it possible to combine this into a single `pipe`?  
3. Ideally, I'd like to use Remeda to compose it into a function that looks like:
    ```ts
    export const toDualCaseObject = R.something
    ```
    How?

Thanks,
your loving boneskull

**Comment by cjquines** [comment, marked as answer]
1: you can replace it with `R.toCamelCase()` (with a call), rather than `R.toCamelCase`, and that should work

2 and 3: don't really think this is super possible... maybe compose `R.fromEntries`, `R.flatMap` and `R.toEntries`? though you might lose types
